client/src/rr/dnssec/signer.rs

```diff
@@ -438,7 +438,7 @@ impl Signer {
             // need a separate encoder here, as the encoding references absolute positions
             // inside the buffer. If the buffer already contains the sig0 RDATA, offsets
             // are wrong and the signature won't match.
-            let mut encoder2: BinEncoder = BinEncoder::with_mode(&mut buf2, EncodeMode::Normal);
+            let mut encoder2: BinEncoder = BinEncoder::with_mode(&mut buf2, EncodeMode::Signing);
```

Comment:
BTW this was introduced in 07d01e5, so if you don't mind rebasing in this PR branch, I can merge these commits.